feat: FAQ 페이지 구현

newminkyung opened the pull request:
## 이슈 번호

<!-- 관련 이슈 번호를 연결 -->

- resolve: #13

## 구현 기능

### Accordion 컴포넌트

<img width="300"  src="https://user-images.githubusercontent.com/80238096/213692219-be667da0-8fab-4f4f-a754-3334b27fac33.png">

```js
<Accordion
  accordions={[
    { id: 1, title: 'Title 1', content: 'Content 1' },
    {
      id: 2,
      title: <>Title 2<p>subTitle</p></>,
      content: <Button>버튼 컴포넌트</Button>,
    },
    { id: 3, title: 'Title 3', content: 'Content 3' },
  ]}
/>
```

| props      | value                                                                                        | 설명                                                       |
| ---------- | -------------------------------------------------------------------------------------------- | ---------------------------------------------------------- |
| accordions | `array` <br>{<br> id: number <br> title: React.ReactNode <br> content: React.ReactNode <br>} | - title, content: 문자열 뿐만 아니라 요소도 들어갈 수 있다 |

### FAQ API

- 성공, 실패 처리 필요

### useFaqAccordion

- FAQ를 최신순으로 정렬
- visible이 true인 FAQ만 필터링
- accordion props 형식에 맞게 반환
  - id, title, content 객체 배열

## 스크린샷

<!--해당 PR에 대한 이미지 -->

![image](https://user-images.githubusercontent.com/80238096/213694343-0cbaba07-6659-4c74-a893-561face1bbae.png)

## 참고 사항

<!-- 궁금한 점, 논의할 점 등 -->

- 이해하기 어려운 부분은 피드백 부탁드립니다🫡

newminkyung committed:
feat: Accordion 컴포넌트 구현

- title: ReactNode
- content: ReactNode

Related to: #13

newminkyung committed:
feat: FAQ API 구현 (#13)

newminkyung committed:
feat: FAQ 쿼리 커스텀 훅 구현 (#13)

- useQuery로 FAQ 요청

Resolve: #13

newminkyung committed:
feat: FAQ Accordion props를 반환하는 커스텀 훅 구현 (#13)

- FAQ를 최신순으로 정렬
- visible이 true인 FAQ만 사용
- accordion props 형식에 맞게 반환

Resolve: #13

newminkyung committed:
feat: FAQ 페이지 구현 (#13)

inyeong-kang reviewed (approved):
FAQ 페이지 만드시느라 고생 많으셨습니다! 몇번 코드리뷰하다 보니까 react query 어떻게 써야할지 감이 오고 있네요😃 감사합니다!!
혹시 Accordion을 molecule로 분리해주셨는데, Accordion 한줄을 하나의 molecule로 보시는건지 아니면 Accordion들을 하나의 molecule로 보시는건지 궁금합니다! (한줄이라고 말했는데 질문 하나-응답 하나 로 표현하는게 정확하겠네요..)

newminkyung commented:
@inyeong-kang 

리뷰보고 다시 생각해보니가 **molecule은 atom의 조합**이라는 일관성이 깨지네요.. 수정하도록 하겠습니다!
질문에 대한 답변은 아래와 같습니다!

---

**Accordion을 하나의 molecule**로 판단했습니다!

Accordion 컴포넌트의 특징은 2가지가 있습니다.
1. 질문과 응답을 쌍으로 가진다
2. 여러개의 질문과 응답을 가질 수 있다.

고민했던 부분이
1. 질문과 응답을 분리해야할까? → `<Accordion.Title>`, `<Accordion.Content>`

분리하면 Accordion을 사용하는 쪽에서 Title, Content를 조립해서 사용해야합니다. 

```js
// Title과 Content로 나누면 아래 코드를 반복해야한다.
<Accordion>
  <Accordion.Title>Title 1</Accordion.Title> // <summary> 부분
  <Accordion.Content>Content</Accordion.Content> // <div> 부분
</Accordion>
```

- 즉, Title, Content가 따로 재사용이 되는 경우가 없다고 판단하여 atom으로 분리하지 않았습니다!

---

이거는 코드 작성했을 때 생각이고 **Title, Content를 atom으로 분리하고 조합해서 사용**하는 것으로 수정하겠습니다!

newminkyung committed:
refactor: Accordion Title, Content atom 컴포넌트 분리

- props명 변경: accordion → details
- HTML 요소명과 동일할수록 가독성이 높다고 생각해서 변경

Related to: #13

newminkyung committed:
Merge branch 'dev' of https://github.com/sos-sejong-opensource-software/seggle-frontend into feat-faq
